charts/portworx/templates/storage-cluster.yaml

```diff
@@ -0,0 +1,163 @@
+{{- if not (lookup "apps/v1" "DaemonSet" "kube-system" "portworx") }}
+
+  {{- $isCoreOS := .Values.isTargetOSCoreOS | default false }}
+  {{- $internalKVDB := .Values.internalKVDB | default false }}
+  {{- $etcdEndPoints := .Values.etcdEndPoint }}
+  {{- $openshiftInstall := .Values.openshiftInstall | default false }}
+  {{- $EKSInstall := .Values.EKSInstall | default false }}
+  {{- $pksInstall := .Values.pksInstall | default false }}
+  {{- $AKSInstall := .Values.AKSInstall | default false }}
+  {{- $usefileSystemDrive := .Values.usefileSystemDrive | default false }}
+  {{- $usedrivesAndPartitions := .Values.usedrivesAndPartitions | default false }}
+  {{- $secretType := .Values.secretType | default "k8s" }}
+  {{- $deployEnvironmentIKS := .Capabilities.KubeVersion.GitVersion | regexMatch "IKS" }}
+  {{- $drives := .Values.drives | default "none" }}
+  {{- $dataInterface := .Values.dataInterface | default "none" }}
+  {{- $managementInterface := .Values.managementInterface | default "none" }}
+  {{- $envVars := .Values.envVars | default "none" }}
+  {{- $customRegistryURL := .Values.customRegistryURL | default "none" }}
+  {{- $registrySecret := .Values.registrySecret | default "none" }}
+  {{- $licenseSecret := .Values.licenseSecret | default "none" }}
+  {{- $kvdbDevice := .Values.kvdbDevice | default "none" }}
+
+kind: StorageCluster
+apiVersion: core.libopenstorage.org/v1
+metadata:
+  name: "{{ required "Clustername cannot be empty" .Values.clusterName }}"
+  namespace: kube-system
+  annotations:
+    {{- if eq $openshiftInstall true }}
+    portworx.io/is-openshift: "true"
+    {{- end }}
+    {{- if eq $pksInstall true }}
+    portworx.io/is-pks: "true"
+    {{- end }}
+    {{- if eq $EKSInstall true }}
+    portworx.io/is-eks: "true"
+    {{- end }}
+    {{- if eq $AKSInstall true }}
+    portworx.io/is-aks: "true"
+    {{- end }}
+    # TODO: remove
+    portworx.io/migration-approved: "false"
+spec:
+  image: portworx/oci-monitor:{{ required "A valid Image tag is required in the SemVer format" .Values.imageVersion }}
+  imagePullPolicy: Always
+  {{- if not (eq $customRegistryURL "none") }}
+  customImageRegistry: {{ $customRegistryURL }}
+  {{- end }}
+  {{- if not (eq $registrySecret "none") }}
+  imagePullSecret: {{ $registrySecret }}
+  {{- end }}
+
+  kvdb:
+  {{- if eq $internalKVDB true }}
+    internal: true
+  {{- else }}
+    internal: false
+    {{- if empty $etcdEndPoints }}
+    "{{ required "A valid ETCD url in the format etcd:http://<your-etcd-endpoint> is required. Verify that the key is correct and there isnt any typo in specifying that, also ensure it is accessible from all node of your kubernetes cluster" .etcdEndPoint}}"
+    {{- else }}
+    endpoints:
+      {{- $endpoints := $etcdEndPoints | split ";" }}
+      {{- range $key, $val := $endpoints }}
+      - {{$val}}
+      {{- end }}
+    {{- end }}
+  {{- end }}
+
+  storage:
```

Comment:
The storage section does not keep the same behavior as the original helm chart. We should keep the behavior consistent and not break backward compatibility wherever possible.
```
              {{- if eq $drives "none" }}
                  {{- if .AKSInstall -}}
                   "-s", "type=Premium_LRS,size=150",
                  {{- else if eq $usedrivesAndPartitions true }}
                      "-A",
                  {{- else }}
                     "-a",
                  {{- end -}}
                 {{- if or $usefileSystemDrive $deployEnvironmentIKS }}
                      "-f",
                  {{- end }}
              {{- else -}}
                {{- $driveNames := $drives | split ";" }}
                {{- range $index, $name := $driveNames }}
                  "-s", "{{ $name }}",
                {{- end -}}
              {{- end -}}
```

Reply:
I reviewed the code when writing this section it was actually handled in operator code, the template should just fill in the value and let operator code handle it all together.